lib/smf/Kconfig

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+config SMF
+	bool "Enable Hierarchical State Machine"
+	help
+	  This option enables the Hierarchical State Machine library
+
+if SMF
+
+config SMF_ANCESTOR_NUM
```

Comment:
I suggest you create a helper symbol:

```
config SMF_ANCESTOR_SUPPORT
    def_bool SMF_ANCESTOR_NUM > 0
    help
       if y, then the state machine framework includes ancestor state support
```

This allows you to replace the references of ``#if (CONFIG_SMF_ANCESTOR_NUM > 0)`` with ``#ifdef CONFIG_SMF_ANCESTOR_SUPPORT`` and ``IS_ENABLED(CONFIG_SMF_ANCESTOR_SUPPORT``.

Reply:
done